RxStarscreamTests/RxStarscreamTests.swift

```diff
@@ -0,0 +1,159 @@
+                    ()
+                }
+                .subscribe(onNext: {
+                    connectExpectation.fulfill()
+                }).addDisposableTo(disposeBag)
+
+        socket.rx.connected
+                .subscribe(connectedObserver)
+                .addDisposableTo(disposeBag)
+
+        waitForExpectations(timeout: 10.0) { error in
+
+            XCTAssertEqual(self.connectedObserver.events.count, 2)
+            XCTAssertEqual(self.connectedObserver.events[0].value.element!, true)
+            XCTAssertEqual(self.connectedObserver.events[1].value.element!, false)
+        }
+    }
+
+    func testPongMessage() {
+        socket.connect()
+
+        socket.rx.connected.subscribe(onNext: { [unowned self] connected in
+                    self.socket.write(ping: Data())
+                }).addDisposableTo(disposeBag)
+
+        let pongExpectation = expectation(description: "Pong")
+        let scheduler = TestScheduler(initialClock: 0)
+        pongObserver = scheduler.createObserver(WebSocketEvent.self)
+
+        socket.rx.response
+                .scan(0, accumulator: { acc, _ in return acc + 1 })
+                .filter {
+                    $0 == 2
+                }
+                .map { _ in
+                    ()
+                }
+                .subscribe(onNext: {
+                    pongExpectation.fulfill()
+                }).addDisposableTo(disposeBag)
+
+        socket.rx.response
+                .subscribe(pongObserver)
+                .addDisposableTo(disposeBag)
+
+        waitForExpectations(timeout: 10.0) { error in
+
+            XCTAssertEqual(self.pongObserver.events.count, 2)
+            XCTAssertEqual(self.pongObserver.events[1].value.element!, WebSocketEvent.pong)
+        }
+    }
+
+    func testMessageResponse() {
+        let sentMessage = "Hello"
+
+        socket.connect()
+
+        socket.rx.connected.subscribe(onNext: { [unowned self] _ in
+                    self.socket.write(string: sentMessage)
+                }).addDisposableTo(disposeBag)
+
+        let messageExpectation = expectation(description: "Message")
+        let scheduler = TestScheduler(initialClock: 0)
+        responseObserver = scheduler.createObserver(WebSocketEvent.self)
+
+        socket.rx.response
+                .scan(0, accumulator: { acc, _ in return acc + 1 })
+                .filter {
+                    $0 == 2
+                }
+                .map { _ in
+                    ()
+                }
+                .subscribe(onNext: {
+                    messageExpectation.fulfill()
+                }).addDisposableTo(disposeBag)
+
+        socket.rx.response
+                .subscribe(responseObserver)
+                .addDisposableTo(disposeBag)
+
+        waitForExpectations(timeout: 10.0) { error in
+
+            XCTAssertEqual(self.responseObserver.events.count, 2)
+            if case WebSocketEvent.message(let message) = self.responseObserver.events[1].value.element! {
```

Comment:
`XCTAssertEqual(message, WebSocketEvent.message(sentMessage))` should be enough here with a proper `==` implementation, it'll make it a bit cleaner than having an `if case` imho